vortex-ipc/src/messages.rs

```diff
@@ -186,15 +195,147 @@ impl<'a> WriteFlatBuffer for IPCArray<'a> {
             .collect_vec();
         let children = Some(fbb.create_vector(&children));
 
-        fba::Array::create(
+        let stats = compute_and_build_stats(fbb, self.1, self.0.stats());
+
+        fb::Array::create(
             fbb,
-            &fba::ArrayArgs {
+            &fb::ArrayArgs {
                 version: Default::default(),
                 has_buffer: column_data.buffer().is_some(),
                 encoding,
                 metadata,
+                stats: Some(stats),
                 children,
             },
         )
     }
 }
+
+/// Computes all stats and uses the results to create an ArrayStats table for the flatbuffer message
+fn compute_and_build_stats<'a>(
+    fbb: &'_ mut FlatBufferBuilder<'a>,
+    array: &'_ ArrayData,
+    to_compute: &[Stat],
+) -> WIPOffset<fb::ArrayStats<'a>> {
+    let primitive_ptype = match array.dtype() {
+        DType::Primitive(ptype, _) => Some(ptype),
+        _ => None,
+    };
+
+    let mut frequencies: HashMap<_, _> = to_compute
+        .iter()
+        .flat_map(|&stat| match stat {
+            Stat::BitWidthFreq => Some((
+                stat,
+                array
+                    .statistics()
+                    .compute_bit_width_freq()
+                    .ok()
+                    .map(|v| v.iter().map(|&inner| inner as u64).collect_vec())
+                    .map(|v| fbb.create_vector(v.as_slice())),
+            )),
+            Stat::TrailingZeroFreq => Some((
+                stat,
+                array
+                    .statistics()
+                    .compute_trailing_zero_freq()
+                    .ok()
+                    .map(|v| v.iter().map(|&inner| inner as u64).collect_vec())
+                    .map(|v| fbb.create_vector(v.as_slice())),
+            )),
+            _ => None,
+        })
+        .flat_map(|(stat, value)| value.map(|v| (stat, v)))
+        .collect();
+
+    let mut counts: HashMap<_, _> = to_compute
```

Comment:
This was to avoid if-contains/else for all stats, not sure if I like this better.